src/telegram_to_engagement_db/telegram_group_to_engagement_db.py

```diff
@@ -0,0 +1,240 @@
+
+    return telegram
+
+
+async def _fetch_message_from_group(telegram, group_id, dataset_end_date=None, dataset_group_latest_seen_message_id=None):
+    """
+    :param telegram: Instance of telegram app to use to download the group messages from.
+    :type telegram: telethon.client.telegramclient.TelegramClient
+    :param group_id: Id of the telegram group to fetch messages from
+    :type group_id: str
+    :param dataset_end_date: Offset datetime, messages previous to this date will be retrieved. Exclusive
+    :type dataset_group_latest_seen_message_id: datetime | None
+    :param dataset_group_latest_seen_message_id: All the messages with a lower (older) ID or equal to this will be excluded.
+    :type group_latest_seen_message_id: int | None
+    :yields: Instances of telethon.tl.custom.message.Message
+    """
+    # Get group/channel entity
+    group_entity = await telegram.get_entity(PeerChannel(int(group_id)))
+
+    #Fetch messages messages based on dataset_offset_date and/or min_id filters if specified.
+    if dataset_end_date is None and dataset_group_latest_seen_message_id is None:
+        log.info(f"Fetching all messages from group {group_id}")
+        return telegram.iter_messages(group_entity)
+
+    elif dataset_end_date is not None and dataset_group_latest_seen_message_id is None:
+        log.info(f"Fetching messages from group {group_id} sent before {dataset_end_date}, exclusive")
+        return telegram.iter_messages(group_entity, offset_date=dataset_end_date)
+
+    elif dataset_end_date is None and dataset_group_latest_seen_message_id is not None:
+        log.info(f"Fetching messages from group {group_id} with message.id greater than "
+                 f"{dataset_group_latest_seen_message_id}, exclusive")
+        return telegram.iter_messages(group_entity, min_id=int(dataset_group_latest_seen_message_id))
+
+    elif dataset_end_date is not None and dataset_group_latest_seen_message_id is not None:
+        log.info(f"Fetching messages from group {group_id} sent before {dataset_end_date} "
+                 f"and with message.id greater than {dataset_group_latest_seen_message_id}, both exclusive")
+        return telegram.iter_messages(group_entity, offset_date=dataset_end_date,
+                                      min_id=int(dataset_group_latest_seen_message_id))
+
+
+def _is_avf_message(telegram_message):
+    """
+    :param telegram_message: A telegram message object to check if it was sent by an admin or is a channel broadcast
+                             Returns True if a message.from_id is None because admins send messages anonymously in the group
+                             and/or is of the type PeerChannel i.e channel broadcast messages.
+    :type telegram_message: telethon.tl.custom.message.Message
+    """
+    # Skip messages sent by AVF group admins / channel broadcasts
+    if (type(telegram_message.from_id) == PeerChannel or telegram_message.from_id is None):
+        return True
```

Comment:
```suggestion
    return (type(telegram_message.from_id) == PeerChannel or telegram_message.from_id is None):
```
so you can still get False back